frontend/src/components/SystemSubmitDrawer/index.tsx

```diff
@@ -57,16 +59,38 @@ export function SystemSubmitDrawer(props: Props) {
   const [otherTargetLang, setOtherTargetLang] = useState(false);
 
   const [form] = useForm<FormData>();
-  const { onClose } = props;
+  const { systemToEdit, onClose, ...rest } = props;
+  const editMode = systemToEdit !== undefined;
+
+  const resetAllFormFields = useCallback(() => {
+    form.resetFields([
+      "name",
+      "task",
+      "dataset",
+      "sys_out_file",
+      "custom_dataset_file",
+      "metric_names",
+      "is_private",
+      "source_language",
+      "target_language",
+      "shared_users",
+      "system_details",
+    ]);
+  }, [form]);
 
   useEffect(() => {
     async function fetchTasks() {
+      setState(State.loading);
       setTaskCategories(await backendClient.tasksGet());
       setLanguageCodes(await backendClient.languageCodesGet());
       setState(State.other);
     }
-    fetchTasks();
-  }, []);
+
+    if (!editMode) {
```

Comment:
nit: I don't think we need to make an exception for `editMode`. We can probably just go ahead and get the tasks list regardless.

Reply:
Thanks, removed the if in favor of simplicity